### DISCUSSION TITLE
Desktop Cruiser GL on Mac

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#20] [Q&A] [answered]
There were reports in the [old forum](https://groups.google.com/g/cruiser-users) that Cruiser with "Settings | Rendering engine | VTM (OpenGL)" not work on Mac (Intel).

For those with a Mac (Intel), can you try running Cruiser with VTM (OpenGL) with the command:
```console
java -XstartOnFirstThread -jar cruiser.jar
```

**Comment by devemux86** [comment]
[Cruiser 3.0.8](https://github.com/devemux86/cruiser/releases) is available with a new shortcut `cruiser-mac.sh` for Mac.

Please try Cruiser GL on Mac (Intel) and see if it works properly.

---

To run Cruiser GL on Mac (Arm) you can try [Apple Rosetta](https://support.apple.com/en-us/HT211861).

"Rosetta 2 enables a Mac with Apple silicon to use apps built for a Mac with an Intel processor."

**Reply by treee111** [reply]
I'd also love to have cruiser with GL on a Mac!
I tried the command as well as the .sh file with the (not suprisingly) same output. The cruiser window neither any Java application has opened:
![Bildschirmfoto 2023-02-03 um 00 12 38](https://user-images.githubusercontent.com/53038537/216472437-54be5564-4671-4192-807a-16f7b93bbc41.png)

Opening cruiser with `cruiser.jar`, change the rendering engine and restart looks like that for me:
![Bildschirmfoto 2023-02-03 um 00 19 00](https://user-images.githubusercontent.com/53038537/216473027-e9540c19-2d69-429d-8305-d7700fe556d4.png)


My machine: MacBook Pro 8,1 (2011) with OCLP patched macOS Ventura 13.2. openJDK

If I should tryout something or can help in any other way, I'm happy to help!

**Reply by devemux86** [reply]
@treee111 thanks for the tests!

So this flag cannot help and something else is needed on Mac (Intel).

I will check it when time is available and maybe create test jars to try.

---

That exception seems from a loaded theme missing some resources.

**Reply by treee111** [reply]
no worries!
Thanks for the hint, you might be right the errors could be from a theme I loaded.

I tested on a M1 MacBookPro17,1 with Monterey 12.6.1 installed.
`cruiser-mac.sh` has the same output as with my other Mac.

After starting cruiser with `cruiser.sh` and then changing the rendering engine in App + restart, this error appears:
<img width="795" alt="Bildschirmfoto 2023-02-03 um 13 53 13" src="https://user-images.githubusercontent.com/53038537/216608680-f7f11941-dddb-452d-971a-d6e1d0056798.png">

On the Apple help page it says one is prompted to install Rosetta 2. For me, there was no prompt and the manuall installed via `softwareupdate --install-rosetta` failed for me (probably because I am not admin of this computer at the moment):
<img width="576" alt="Bildschirmfoto 2023-02-03 um 14 00 35" src="https://user-images.githubusercontent.com/53038537/216609985-dd114683-9f65-4501-9a8e-ac5f697ffafa.png">

dont worry, there is no pressure here for me!

**Reply by devemux86** [reply, marked as answer]
There are 2 different cases for Mac depending on the CPU:
- Mac with **Intel**
Cruiser GL uses [VTM](https://github.com/mapsforge/vtm) OpenGL map library.
Users report that [vtm-playground](https://github.com/mapsforge/vtm/blob/master/docs/desktop.md) sample runs on Mac (Intel), but not Cruiser GL (much more complex app).
So research and testing on Cruiser GL is needed...
- Mac with **ARM**
Nothing runs, first someone must build VTM + LWJGL 2 native libraries for Mac (ARM).
See https://github.com/mapsforge/vtm/issues/1004
Or [Apple Rosetta](https://support.apple.com/en-us/HT211861) could probably help this case.

I work on Linux, so I cannot do much testing on Windows or Mac. :slightly_smiling_face:

**Reply by uhagi** [reply]
Any news on this ?
Also trying to run cruiser on Mac OS 12.6. No success with vim. Works perfect with maps forge renderer.

I have installed OpenGL via homebrew. (glfw3 and glew). test-app seems to find OpenGL in version 4.1. But cruiser still doesnt show anything in map area in vim-mode...

**Reply by devemux86** [reply]
@uhagi 

VTM needs help from a developer with Mac to build the VTM native libraries for Max with ARM.
- https://github.com/mapsforge/vtm/issues/1004

Otherwise it is better to use the Mapsforge renderer that runs everywhere.

**Reply by uhagi** [reply]
Ah sorry. I have a x86 machine. I would like to use the vtm-renderer in order to modify and test certain vtm-renderer-themes...

**Reply by devemux86** [reply]
@uhagi

Users mention that Cruiser (OpenGL) works on Mac with x86.
Unless OpenGL is not working properly.

**Comment by devemux86** [comment]
You can make Cruiser start again with Mapsforge with the command:

```console
java -jar cruiser.jar mapsforge
```
